File: backend/Models/Prescription.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Models
{
    public class Prescription
    {
        public int Id { get; set; }

        [Required]
        public int PatientId { get; set; }

        public Patient? Patient { get; set; }

        [Required]
        public DateTime PrescriptionDate { get; set; }

        [Required]
        public string PrescriptionType { get; set; } = string.Empty;

        public string RightEyeSphere { get; set; } = string.Empty;
        public string LeftEyeSphere { get; set; } = string.Empty;

        public string RightEyeCylinder { get; set; } = string.Empty;
        public string LeftEyeCylinder { get; set; } = string.Empty;

        public string RightEyeAxis { get; set; } = string.Empty;
        public string LeftEyeAxis { get; set; } = string.Empty;

        public string PupillaryDistance { get; set; } = string.Empty;

        public string MedicationName { get; set; } = string.Empty;
        public string Dosage { get; set; } = string.Empty;
        public string Instructions { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; } = DateTime.Now;
    }
}

File: frontend/src/app/models/prescription.ts
import { Patient } from './patient';

export interface Prescription {
  id: number;
  patientId: number;
  patient?: Patient;
  prescriptionDate: string;
  prescriptionType: string;
  rightEyeSphere: string;
  leftEyeSphere: string;
  rightEyeCylinder: string;
  leftEyeCylinder: string;
  rightEyeAxis: string;
  leftEyeAxis: string;
  pupillaryDistance: string;
  medicationName: string;
  dosage: string;
  instructions: string;
  createdAt: string;
}

export interface CreatePrescription {
  patientId: number;
  prescriptionDate: string;
  prescriptionType: string;
  rightEyeSphere: string;
  leftEyeSphere: string;
  rightEyeCylinder: string;
  leftEyeCylinder: string;
  rightEyeAxis: string;
  leftEyeAxis: string;
  pupillaryDistance: string;
  medicationName: string;
  dosage: string;
  instructions: string;
}